rocket_twin/systems/tank/pipe.py

```diff
@@ -0,0 +1,31 @@
+from cosapp.base import System
+
+
+class Pipe(System):
+    """A simple model of a pipe.
+
+    Inputs
+    ------
+    w_in [kg/s]: float,
+        mass flow of fuel entering the pipe
+    is_open: boolean,
+        whether the pipe is open or closed
+
+    Outputs
+    ------
+    w_out [kg/s]: floatm
+        mass flow of fuel exiting the pipe
+    """
+
+    def setup(self):
+
+        self.add_inward("is_open", False, desc="Whether the pipe is open or not", unit="")
+        self.add_inward("w_in", 0.0, desc="Fuel income rate", unit="kg/s")
+
+        self.add_outward("w_out", 0.0, desc="Fuel exit rate", unit="kg/s")
+
+    def compute(self):
+
+        if self.is_open is False:
+            self.parent.g_tank.w_out_temp = 0.0
```

Comment:
it is forbeten to change a parent value in a system. more over, it is system architecture dependant. you must manage pipe opening a different way.

Reply:
Changed the "valve" and the control to the tank system